spec/models/user_action_spec.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe UserAction, type: :model do
+  describe 'validations' do
+    let(:acting_ip_address) { Faker::Internet.ip_v4_address }
+    let(:acting_user_agent) { Faker::Internet.user_agent }
+    let(:acting_user_verification) { build(:user_verification) }
+    let(:user_action) do
+      build(:user_action,
+            acting_ip_address:,
+            acting_user_agent:,
+            acting_user_verification:)
+    end
+
+    context 'when attributes are valid' do
+      it 'is valid' do
+        expect(user_action).to be_valid
+      end
+    end
+
+    context 'when acting_ip_address is nil' do
+      let(:acting_ip_address) { nil }
+
+      it 'is valid' do
+        expect(user_action).to be_valid
+      end
+    end
+
+    context 'when acting_user_agent is nil' do
+      let(:acting_user_agent) { nil }
+
+      it 'is valid' do
+        expect(user_action).to be_valid
+      end
+    end
+
+    context 'when acting_user_verification is nil' do
+      let(:acting_user_verification) { nil }
+
+      it 'is valid' do
+        expect(user_action).to be_valid
+      end
+    end
```

Comment:
I don’t think it’s necessary to test attribute validations unless they are explicitly defined in the model. These will be valid regardless of whether they are `nil` or not.